peer/direct/direct_test.go

```diff
@@ -115,3 +115,9 @@ func TestDirect(t *testing.T) {
 		onFinish(nil)
 	})
 }
+
+func TestPeerSubscriber(t *testing.T) {
```

Comment:
can we add a comment why we are asserting in the address equality or simply rename the test to be: TestPeerSubscriberAsHashKey?